src/main/java/org/wildfly/security/auth/client/SetPasswordAuthenticationConfiguration.java

```diff
@@ -62,6 +63,10 @@ void handleCallback(final Callback[] callbacks, final int index) throws IOExcept
             } catch (GeneralSecurityException e) {
                 // fall out
             }
+        } else if (callback instanceof RealmCallback) {
+            RealmCallback realmCallback = (RealmCallback) callback;
+            realmCallback.setText(realmCallback.getDefaultText());
```

Comment:
At the first I has think this is useless, because DigestSaslClient use RealmCallback only to inform handler about used realm (handler can select username+password by realm). But when I checked it in different SaslClients, this line is realy required. (Maybe it is bug of DigestSaslClient when it ignore output of RealmCallback.)